### PR TITLE
swap functions

### PR DESCRIPTION
**Pull request by kanake10**
pr showing how to write swap functions in Kotlin

### CODE REVIEW

File: core-kotlin-modules/core-kotlin-7/src/main/kotlin/com/baeldung/swapfunction/SwapWithToOperator.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.baeldung.swapfunction
+
+fun <T> swap(a: T, b: T): Pair<T, T> {
+    return Pair(b, a)
+}
+
+class SwapWithToOperator {
```
Comment (Bwvolleyball) [inline core-kotlin-modules/core-kotlin-7/src/main/kotlin/com/baeldung/swapfunction/SwapWithToOperator.kt:7]:
```suggestion
class SwapWithPair {
```

Comment (kanake10) [reply core-kotlin-modules/core-kotlin-7/src/main/kotlin/com/baeldung/swapfunction/SwapWithToOperator.kt]: on it!!

File: core-kotlin-modules/core-kotlin-7/src/test/kotlin/com/baeldung/swapfunction/classes/SwapValue.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        assertEquals(2, b)
+        val temp = a
+        a = b
+        b = temp
+        assertEquals(2, a)
+        assertEquals(1, b)
+    }
+}
+
+
```
Comment (Bwvolleyball) [inline core-kotlin-modules/core-kotlin-7/src/test/kotlin/com/baeldung/swapfunction/classes/SwapValue.kt:20]: We'll need a test import for this annotations
```suggestion
class SwapValue {

    @Test
    fun swapValues(){
        var a = 1
        var b = 2
        assertEquals(1, a)
        assertEquals(2, b)
        val temp = a
        a = b
        b = temp
        assertEquals(2, a)
        assertEquals(1, b)
    }
}

```

File: core-kotlin-modules/core-kotlin-7/src/test/kotlin/com/baeldung/swapfunction/classes/SwapWithAlso.kt [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    fun swapWithAlso(){
+
+        assertEquals(1, a)
+        TestCase.assertEquals(2, b)
+        val temp = a
+        a = b
+        b = temp
+        assertEquals(2, a)
+        assertEquals(1, b)
+    }
```
Comment (Bwvolleyball) [inline core-kotlin-modules/core-kotlin-7/src/test/kotlin/com/baeldung/swapfunction/classes/SwapWithAlso.kt:18]:
```suggestion
    @Test
    fun swapWithAlso(){
        var a = 1
        var b = 2

        assertEquals(1, a)
        assertEquals(2, b)
        val temp = a
        a = b
        b = temp
        assertEquals(2, a)
        assertEquals(1, b)
    }
```

File: core-kotlin-modules/core-kotlin-7/src/test/kotlin/com/baeldung/swapfunction/classes/SwapWithPair.kt [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    fun swapUsingDestructuring(){
+        assertEquals(100, x)
+        assertEquals(200, y)
+        val result = swap(x, y)
+        x = result.first
+        y = result.second
+        assertEquals(200, x)
+        assertEquals(100, y)
+    }
+}
```
Comment (Bwvolleyball) [inline core-kotlin-modules/core-kotlin-7/src/test/kotlin/com/baeldung/swapfunction/classes/SwapWithPair.kt:23]:
```suggestion
    @Test
    fun swapUsingDestructuring(){
        var x = 100
        var y = 200
        assertEquals(100, x)
        assertEquals(200, y)
        val result = swap(x, y)

        x = result.first
        y = result.second
        assertEquals(200, x)
        assertEquals(100, y)
    }
}

```

File: core-kotlin-modules/core-kotlin-7/src/test/kotlin/com/baeldung/swapfunction/tests/SwapWithPairTest.kt
Comment (Bwvolleyball) [inline core-kotlin-modules/core-kotlin-7/src/test/kotlin/com/baeldung/swapfunction/tests/SwapWithPairTest.kt:1]: This file can be replaced if we convert the `SwapWithPair` class to contain the test

File: core-kotlin-modules/core-kotlin-7/src/test/kotlin/com/baeldung/swapfunction/tests/SwapWithAlsoTest.kt
Comment (Bwvolleyball) [inline core-kotlin-modules/core-kotlin-7/src/test/kotlin/com/baeldung/swapfunction/tests/SwapWithAlsoTest.kt:1]: This file can also be replaced if we just have the test in `SwapWithAlso`

File: core-kotlin-modules/core-kotlin-7/src/test/kotlin/com/baeldung/swapfunction/tests/SwapValueTest.kt
Comment (Bwvolleyball) [inline core-kotlin-modules/core-kotlin-7/src/test/kotlin/com/baeldung/swapfunction/tests/SwapValueTest.kt:1]: This file can also be replaced if we make `SwapValue` a test.

File: core-kotlin-modules/core-kotlin-7/src/test/kotlin/com/baeldung/swapfunction/classes/SwapWithPair.kt [hunk 5]
```diff
@@ -0,0 +1,23 @@
+package com.baeldung.swapfunction.classes
+
+
+import junit.framework.TestCase.assertEquals
```
Comment (Bwvolleyball) [inline core-kotlin-modules/core-kotlin-7/src/test/kotlin/com/baeldung/swapfunction/classes/SwapWithPair.kt:4]:
```suggestion
import org.junit.Assert.assertEquals
import org.junit.Test
```

File: core-kotlin-modules/core-kotlin-7/src/test/kotlin/com/baeldung/swapfunction/classes/SwapWithAlso.kt [hunk 6]
```diff
@@ -0,0 +1,19 @@
+package com.baeldung.swapfunction.classes
+
+import junit.framework.TestCase
+import junit.framework.TestCase.assertEquals
```
Comment (Bwvolleyball) [inline core-kotlin-modules/core-kotlin-7/src/test/kotlin/com/baeldung/swapfunction/classes/SwapWithAlso.kt:4]:
```suggestion
import org.junit.Assert.assertEquals
import org.junit.Test
```

File: core-kotlin-modules/core-kotlin-7/src/test/kotlin/com/baeldung/swapfunction/classes/SwapValue.kt [hunk 7]
```diff
@@ -0,0 +1,20 @@
+package com.baeldung.swapfunction.classes
+
+import junit.framework.TestCase.assertEquals
```
Comment (Bwvolleyball) [inline core-kotlin-modules/core-kotlin-7/src/test/kotlin/com/baeldung/swapfunction/classes/SwapValue.kt:3]:
```suggestion
import org.junit.Assert.assertEquals
import org.junit.Test
```

**Review by Bwvolleyball — COMMENTED** [review]
Some final cleanup!

**Pull request merged**